### ISSUE TITLE
Rails 5 compatibility

### ISSUE THREAD

**Issue by avdmatamoros** [globalize/globalize#473]
attempting to run globalize on a rails5 api app I get:

    'require': cannot load such file -- active_record/serializers/xml_serializer (LoadError)
    ...
    from .../gems/globalize-4.0.3/lib/patches/active_record/xml_attribute_serializer.rb:1:in `<top (required)>'
    ...
    from .../gems/globalize-4.0.3/lib/globalize.rb:2:in `<top (required)>'

It seems active_record/serializers/xml_serializer will no longer be available for rails 5 (at least for APIs), which will cause issues once it gets published.

**Comment by lessan** [comment]
I also got a similar error when attempting to migrate to Rails 5. It was because `active_record/serializers/xml_serializer` uses `alias_method_chain`, which is deprecated in Rails 5. Some more info here: http://www.justinweiss.com/articles/rails-5-module-number-prepend-and-the-end-of-alias-method-chain/

`alias_method_chain` is used in three places in globalize, and also twice in the globalize-versioning gem. We'll need to update them to use module#prepend and drop Ruby < 2.0 support to be able to use globalize with Rails 5.

**Comment by batamire** [comment]
I can't do `bundle update` to Rails 5.0.0.beta1...

`gem 'globalize', github: 'globalize/globalize'`

```
Bundler could not find compatible versions for gem "activemodel":
  In Gemfile:
    globalize was resolved to 5.1.0, which depends on
      activemodel (< 4.3, >= 4.2.0)

    rails (= 5.0.0.beta1) was resolved to 5.0.0.beta1, which depends on
      activemodel (= 5.0.0.beta1)
```

**Comment by bbnnt** [comment]
bundle update will let you install with rails5, but it will return the error mentionned in the title

**Comment by scott** [comment]
+1

**Comment by juseveno** [comment]
same here
`Bundler could not find compatible versions for gem "activemodel":
  In snapshot (Gemfile.lock):
    activemodel (= 5.0.0.beta2)

  In Gemfile:
    globalize was resolved to 5.1.0, which depends on
      activemodel (< 4.3, >= 4.2.0)
`

**Comment by monkseal** [comment]
Does anybody have a fork out there that will work with rails 5?

**Comment by tomash** [comment]
There's already a work-in-progress PR #485 , I'll be happy to merge when it's done and green.

**Comment by vizo** [comment]
Works everything just fallback doesn't. Did configuration changed?

**Comment by rhymes** [comment]
@parndt any chance this patch can be loaded only if XmlSerializer is actually there? https://github.com/globalize/globalize/blob/master/lib/patches/active_record/xml_attribute_serializer.rb

It's not there in Rails 5 by default anymore.

See https://github.com/rails/rails/pull/21161 and https://github.com/rails/activemodel-serializers-xml

Thanks!

**Comment by parndt** [comment]
Can you fix it please? :smile:

**Comment by rhymes** [comment]
@parndt https://github.com/globalize/globalize/pull/524

**Comment by rhymes** [comment]
@parndt I tried in #524 but there are three more issues:

- travis keep building Gemfile.rails5 with Ruby 2.1 and 2.0 which results in an error since Rails 5 does not support Ruby < 2.2.2 (see https://travis-ci.org/globalize/globalize/builds/141650459)

- the actual build with Rails 5 has a circular dependency (which in my app results in a stack too deep error), I think it's due to the file https://github.com/globalize/globalize/blob/master/lib/patches/active_record/rails5/uniqueness_validator.rb and its first line. (see https://travis-ci.org/globalize/globalize/jobs/141650460#L207)

- the build breaks anyway because ActiveRecord cannot quote an array: https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/abstract/quoting.rb#L177 (see https://travis-ci.org/globalize/globalize/jobs/141650460#L235)

**Comment by muhammet** [comment]
+1

**Comment by parndt** [comment]
Fixes for all those were merged already, are you still having trouble?

**Comment by vizo** [comment]
Yes.


Bundler could not find compatible versions for gem "activerecord":
  In Gemfile:
    globalize (~> 5.0.0) was resolved to 5.0.0, which depends on
      activerecord (< 4.3, >= 4.2.0)

    rails (= 5.0.0) was resolved to 5.0.0, which depends on
      activerecord (= 5.0.0)

**Comment by neyp** [comment]
It still depends on activerecord < 4.3
